### ISSUE TITLE
How to put into .git/hooks/pre-commit ?

### ISSUE THREAD

**Issue by danschumann** [uber/npm-shrinkwrap#45]
Hey I put `npm shrinkwrap` into a pre-commit git hook, but either git does its check too soon, or shrinkwrap writes it too late, because that commit doesn't have a changed shrinkwrap.

I'd like to automatically shrinkwrap before a commit and have the changed shrinkwrap file go into that same commit.  Is this possible or do I need to write a bash script?

**Comment by peterjwest** [comment]
I'd like to try this, did you remember to run `git add` in your hook?

**Comment by jcorbin** [comment]
The following suffices:
```
#!/bin/sh
set -e

if ! git diff --cached --quiet -- package.json; then
    npm shrinkwrap # or npm run ... to taste
    git add npm-shrinkwrap.json
fi
```

Tested in a trivial foo repo, it:
- only runs shrinkwrap when package.json changes
- adds the wrote shrinkwrap

**Comment by peterjwest** [comment]
Yeah I just did the same, works fine! If you leave out the `package.json` check, though, it will detect modules which are not saved in your package.json.

**Comment by turadg** [comment]
@jcorbin thanks for that snippet!

I used it to make this drop-in hook code https://github.com/turadg/npm-shrinkwrap-git-hooks
